src/PhpLeagueAccessTokenProvider.php

```diff
@@ -57,13 +60,13 @@ class PhpLeagueAccessTokenProvider implements AccessTokenProvider
      * @param array $scopes
      * @param array $allowedHosts
      */
-    public function __construct(TokenRequestContext $tokenRequestContext, array $scopes = [], array $allowedHosts = [])
+    public function __construct(TokenRequestContext $tokenRequestContext, array $scopes = [], array $allowedHosts = [], ?GenericProvider $oauthProvider = null)
```

Comment:
```suggestion
    public function __construct(TokenRequestContext $tokenRequestContext, array $scopes = [], array $allowedHosts = [], ?AbstractProvider $oauthProvider = null)
```
Could we change this type to `AbstractProvider`?